### PR TITLE
Add SSM model and Get/PutParameter examples

### LINKED ISSUE #383: Add SSM
https://github.com/awslabs/aws-sdk-rust/issues/22

https://github.com/awslabs/smithy-rs/wiki/Adding-support-for-a-new-service

### PR DESCRIPTION
**Pull request by jdisanti**
This fixes #383.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws/sdk/examples/ssm-put-parameter/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+use ssm::model::ParameterType;
+use ssm::{Client, Config, Region};
+
+use aws_types::region::{EnvironmentProvider, ProvideRegion};
+
+#[tokio::main]
+async fn main() {
+    // Determine the region from environment variables or default to us-east-1
+    let region = EnvironmentProvider::new()
```
Comment (rcoh) [inline aws/sdk/examples/ssm-put-parameter/src/main.rs:16]: Can you used `region::default_provider()`? Slightly more future proof

File: aws/sdk/examples/ssm-put-parameter/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        .region()
+        .unwrap_or_else(|| Region::new("us-east-1"));
+
+    // Construct a client
+    let config = Config::builder().region(region).build();
+    let client = Client::from_conf(config);
+
+    // Put a SSM application parameter named `test_parameter_name`
+    match client
```
Comment (rcoh) [inline aws/sdk/examples/ssm-put-parameter/src/main.rs:25]: can you call get_parameter after this and retrieve it?

Comment (rcoh) [reply aws/sdk/examples/ssm-put-parameter/src/main.rs:25]: (partially just to make sure that retrieval works)

File: aws/sdk/examples/ssm-put-parameter/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    // Determine the region from environment variables or default to us-east-1
+    let region = EnvironmentProvider::new()
+        .region()
+        .unwrap_or_else(|| Region::new("us-east-1"));
+
+    // Construct a client
+    let config = Config::builder().region(region).build();
+    let client = Client::from_conf(config);
+
+    // Put a SSM application parameter named `test_parameter_name`
```
Comment (rcoh) [inline aws/sdk/examples/ssm-put-parameter/src/main.rs:24]:
```suggestion
    // Put an SSM application parameter named `test_parameter_name`
```

**Review by rcoh — APPROVED** [review]
LGTM! Just a couple small tweaks

**Comment by rcoh** [comment]
ah, looks like clippy is upset about some names in SSM 🤷🏻

File: aws/sdk/examples/ssm-put-parameter/src/main.rs (hunk 2 above)
Comment (jdisanti) [reply aws/sdk/examples/ssm-put-parameter/src/main.rs:25]: Added the following immediately after the put to test get:
```
    match client
        .get_parameter()
        .name("test_parameter_name")
        .send()
        .await
    {
        Ok(response) => {
            println!("Success! Response: {:?}", response)
        }
        Err(error) => {
            println!("Got an error getting the parameter: {}", error);
            process::exit(1);
        }
    }
```

And it works:
```
Success! Response: GetParameterOutput { parameter: Some(Parameter { name: Some("test_parameter_name"), r#type: Some(String), value: Some("some_value"), version: 4, selector: None, source_result: None, last_modified_date: Some(Instant { seconds: 1621441997, subsecond_nanos: 703999996 }), arn: Some("<omitted>/test_parameter_name"), data_type: Some("text") }) }
```

**Review by rcoh — APPROVED** [review]
LGTM! After merging, you can comment on the upstream issue noting that it will launch in the next release

**Pull request merged**